src/main/scala/ai/privado/exporter/HttpConnectionMetadataExporter.scala

```diff
@@ -41,22 +41,63 @@ class HttpConnectionMetadataExporter(cpg: Cpg, ruleCache: RuleCache) {
   private val STRING_CONTAINS_TWO_SLASH = ".*/.*/.*"
   private val SPRING_APPLICATION_BASE_PATH =
     "(?i)(server[.]servlet[.]context-path|server[.]servlet[.]contextPath)|(spring[.]application[.]name)"
+  private val URL_PATH_WITH_VARIABLE_SYMBOLS                       = "^(?=.*/)(?!.*/$)[${}/\"'a-zA-Z0-9:.,%?_=]+"
+  private val NON_ALPHANUMERIC                                     = "[^a-zA-Z0-9]"
+  private val STRING_WITH_CONSECUTIVE_DOTS_OR_DOT_SLASH_OR_NEWLINE = "(?s).*(\\.\\.|\\./|\n).*"
+  private val ESCAPE_STRING_SLASHES                                = "(\\\")"
+  private val IMPORT_REGEX_WITH_SLASHES                            = "(?s)^(?=.*/)(?!.*/$).*"
 
   private val LAMBDA_SERVERLESS_BASE_PATH = "service"
   private val LAMBDA_SERVERLESS_FILE_NAME = ".*serverless.yml"
 
-  def getEgressUrls = {
+  def getLiteralsFromLanguageFiles: List[String] = {
+    val egressLiterals = cpg
+      .literal(URL_PATH_WITH_VARIABLE_SYMBOLS)
+      .filter(node => node.code.replaceAll(NON_ALPHANUMERIC, "") != "")
+      .filter(!_.code.matches(STRING_WITH_CONSECUTIVE_DOTS_OR_DOT_SLASH_OR_NEWLINE))
+      .inCall
+      .map(node =>
+        node.argument
+          .map(arg => {
+            if node.name == "require" then ""   // skip import statement
+            else if arg.isLiteral then arg.code // collect literal
+            // const loginPath = "api/v1" + "/login" --- Addition Case(javascript, python, java)
+            // const signupPath = `api/v1/${signup}` --- Format String Case for javascript(similar applicable for python, java)
+            else if node.name.indexOf("addition") > -1 || node.name.indexOf("formatString") > -1 then arg.code
```

Comment:
I guess here you are checking if the call node is a `addition` or `formatString`, an optimized way would be
`node.name.equals("<operator>.addition")` and similarly for formatString. Because I believe the time complexity of `equals` is better than `indexOf`